classes/constellation.php

```diff
@@ -50,15 +50,87 @@ public function render_incidents($future=false, $offset=0, $limit = 5, $admin =
     $show = !$future && $incidents["more"];
 
     $offset += $limit;
-
+    $time_reference = new DateTime('NOW');
+    $date_previous_stored = null;
+    $date_incident_previous = null;
+    $incident_header_date = null;
+
+    // Check to see if call is coming from the admin page. If loaded from the admin page
+    // the date headers will not be rendered.
+    // TODO: Should be an eaiser way to determine this...
+    $is_on_admin = false;
+    // Handles default page load
+    $arr_url = explode("/", $_SERVER['PHP_SELF']);
+    if ( 'admin' == strtolower($arr_url[count($arr_url)-2]) ) {
+      $is_on_admin = true;
+    }
+    // Handles calls that comes from the "Load More" button.
+    if (isset( $_SERVER['HTTP_REFERER']) ) {
+      $arr_url = explode("/", $_SERVER['HTTP_REFERER']);
+      if ( 'admin'== strtolower($arr_url[count($arr_url)-2]) ) {  // Get last array
+        $is_on_admin = true;  // We are on the admin dashboard, so prevent rendering of incident dates headers
+      }
+    }
+    // Loop over dataset and create missing dates
     if (count($incidents["incidents"])){
+
+       // Get previous date from last loaded incident via GET to indicate where the rendering starts from
+       if (isset($_GET['lastviewed']) && is_numeric($_GET['lastviewed']) ) {
+          $time_reference->setTimestamp($_GET['lastviewed']);
+          $date_incident_previous =  $time_reference->format("Y-m-d");
+          $date_previous_stored = $date_incident_previous;
+       }
+
       foreach ($incidents['incidents'] as $incident) {
-        $incident->render($admin);
+
+        // Check if first incident is this date or earlier
+        // If earlier -> calc number of days and add no-incident display for timeperiod leading up to
+	
+        $date_incident = new DateTime();
+        $date_incident->setTimestamp($incident->timestamp);
+        $days = $time_reference->diff($date_incident)->format('%a');
+        if ( $days > 1 ) {
+          for ($i = 0; $i <= $days; $i++) {
+            $subs = "-" . $i . " days";
+            $time_reference_clone = clone $time_reference;	// Clone object to avoid master object to change during processing
```

Comment:
can't you skip the cloning and just do `-1 day` on the `$time_reference` every iteration?